helpers/createAgTokenSideChainMultiBridge.sh

```diff
@@ -0,0 +1,144 @@
+  echo -e "totalLimit: total limit of the bridge"
+  echo -e "hourlyLimit: hourly limit of the bridge"
+  echo -e "chainTotalHourlyLimit: chain total hourly limit"
+  echo -e "mock: mock deployment (true/false)"
+  echo -e "expectedAddress: expected address for the token (optional)"
+  echo ""
+}
+
+function main {
+    if [[ $# -ne 6 && $# -ne 7 ]]; then
+        usage
+        exit 1
+    fi
+    chain=$1
+    stableName=$2
+    totalLimit=$3
+    hourlyLimit=$4
+    chainTotalHourlyLimit=$5
+    mock=$6
+    expectedAddress=$7
+
+    if [ -z "$chain" ] || [ -z "$stableName" ] || [ -z "$totalLimit" ] || [ -z "$hourlyLimit" ] || [ -z "$chainTotalHourlyLimit" ] || [ -z "$mock" ] ; then
+        echo "Missing arguments"
+        exit 1
+    fi
+
+    if [ ! -f .env ]; then
+        echo ".env not found!"
+        exit 1
+    fi
+    source .env
+
+    chainName=$(chain_to_name $chain)
+    chainUri=$(chain_to_uri $chain)
+    chainId=$(chain_to_chainId $chain)
+    if [[ -z "$chainUri" || -z "$chainName" || -z "$chainId" ]]; then
+        echo "Invalid chain"
+        exit 1
+    fi
+    # Check if totalLimit is a positive or null integer
+    if ! [[ $totalLimit =~ ^[0-9]+$ ]]; then
+        echo "Total limit must be a positive integer"
+        exit 1
+    fi
+    # Check if hourlyLimit is a positive or null integer
+    if ! [[ $hourlyLimit =~ ^[0-9]+$ ]]; then
+        echo "Hourly limit must be a positive integer"
+        exit 1
+    fi
+    # Check if chainTotalHourlyLimit is a positive or null integer
+    if ! [[ $chainTotalHourlyLimit =~ ^[0-9]+$ ]]; then
+        echo "Chain total hourly limit must be a positive integer"
+        exit 1
+    fi
+    # check if mock is a boolean
+    if [[ "$mock" != "true" && "$mock" != "false" ]]; then
+        echo "Mock must be true or false"
+        exit 1
+    fi
+
+    mainnet_uri=$(chain_to_uri 1)
+
+    if [[ ! -z "$expectedAddress" ]]; then
+       export EXPECTED_ADDRESS=$expectedAddress
+    fi
+    if [[ "$mock" == "true" ]]; then
+        export MOCK=true
+    fi
+
+    export CHAIN_ID=$chainId
+    export TOTAL_LIMIT=$totalLimit
+    export HOURLY_LIMIT=$hourlyLimit
+    export CHAIN_TOTAL_HOURLY_LIMIT=$chainTotalHourlyLimit
+    export STABLE_NAME=$stableName
+
+    echo ""
+    echo "Running deployment on chain $chain for stable token $stableName"
+
+    cd lib/angle-tokens && MNEMONIC_MAINNET=$MNEMONIC_MAINNET forge script DeployAgTokenSideChainMultiBridge --fork-url $chainUri --verify --broadcast && cd ../..
+
+    if [ $? -ne 0 ]; then
+        echo ""
+        echo "Deployment failed"
+        exit 1
+    fi
+
+    echo ""
+    echo "Deployment successful"
+
+    echo ""
+    echo "Would you like to create the multisig transaction for the agToken side chain multi bridge ? (yes/no)"
+
+    read createTransaction
+
+    if [[ $createTransaction == "yes" ]]; then
+
+        forge script ConnectAgTokenSideChainMultiBridge --fork-url $mainnet_uri
```

Comment:
wrong fork-url

Reply:
the fork url can stay mainnet as we use fork when wrapping the transaction

Reply:
it doesn't really matters in this case